### DISCUSSION TITLE
Can't figure out if there is an easy way to go ahead and replace the vim window that spawned the preview.

### DISCUSSION THREAD

**Discussion by unphased** [rmagatti/goto-preview#98] [Q&A]
I love how I can use all my existing binds to resize and cycle focus around to the preview windows, because they're real windows, as well as the ability to use the `<c-w>H/J/K/L` to "pull" them out into bona fide windows as splits!

But... sometimes (around half the time?) I open up a definition and realize that I did want the base operation of just opening the definition in the originating buffer... I'm not interested in having two separate base binds, I think I want to stick to `gd` opening up a preview with this plugin every time. 

It's not clear to me how to add a bind into the preview to make this happen. Presumably I would want to do something like swap the preview window with the last focused window and then kill the preview window. Or something like that.

I'm sure some simple VimL or Lua scripting could get this done efficiently, but I cannot really figure out an elegant way to do this. Could you help me with that?

Here's how I'm already configuring this plugin via Lazy and I love everything else about it. 

```lua
  {
    'rmagatti/goto-preview',
    init = function()
      require('goto-preview').setup {
        default_mappings = false,
        post_open_hook = function(bufnr)
          vim.cmd("setlocal winblend=10")
          vim.keymap.set('n', '<Esc>', '<cmd>q<CR>', { buffer = bufnr, silent = true } )
          vim.keymap.set('n', 'q', '<cmd>q<CR>', { buffer = bufnr, silent = true } )
        end,
      }
    end
  },
```

**Comment by rmagatti** [comment]
I'm actually not sure how you'd do this either. If there was only one window before you could move the float to a split then hit `<C-w>o` to close all other windows. But if you were coming from a split, I'm not sure how you'd replace the right window with the floating window

**Reply by unphased** [reply]
FYI I use Glance now. It is not as pretty, but I haven't explored whether it has customization to improve that. But what is nice about it is by default it opens the overlay window along with a menu from which you can type enter and do the thing i described above.

**Comment by TylerBarnes** [comment]
This is probably not the most ideal way to do it but it works!
In `post_open_hook` I set a keybind that gets the current cursor position, then closes all preview windows, then immediately opens the bufnr passed into `post_open_hook` and set the cursor position. This seems to work because when you close the preview window your cursor goes back to the window it was in when you opened the preview.

```lua
        post_open_hook = function(buf)
          local function set_cursor_position(line_number, column_number)
            local buffer_id = vim.api.nvim_get_current_buf()
            local line_count = vim.api.nvim_buf_line_count(buffer_id)
            if line_number > 0 and line_number <= line_count then
              vim.api.nvim_win_set_cursor(0, { line_number, column_number })
            else
              print("Invalid line number: " .. line_number)
            end
          end

          local function open_buffer_and_jump_to_line(buffer_id, line_number, column_number)
            if vim.api.nvim_buf_is_valid(buffer_id) then
              vim.api.nvim_set_current_buf(buffer_id)
              set_cursor_position(line_number, column_number)
            else
              print("Invalid buffer ID: " .. buffer_id)
            end
          end

          local function get_current_cursor_position()
            local cursor_pos = vim.api.nvim_win_get_cursor(0)
            local line_number = cursor_pos[1]
            local column_number = cursor_pos[2]
            return line_number, column_number
          end

          vim.keymap.set("n", "go", function()
            local line, column = get_current_cursor_position()
            require("goto-preview").close_all_win()
            open_buffer_and_jump_to_line(buf, line, column)
          end)
        end
```

**Reply by TylerBarnes** [reply]
So this does work but it only works if you only open one level of preview. Seems if you open nested previews it ends up putting your cursor back in the leftmost window when you close all previews. I'll probably try to figure out how to prevent that eventually but for now this is a fine limitation for me

**Reply by TylerBarnes** [reply]
Another update, looks like setting `stack_floating_preview_windows = false` is a valid workaround for the limitation I mentioned
